R/get_paths.R

```diff
@@ -57,26 +58,85 @@ get_levelpaths <- function(x, override_factor = NULL, status = FALSE) {
   x[["topo_sort"]] <- seq(nrow(x), 1)
   x[["levelpath"]] <- rep(0, nrow(x))
 
-  flc <- x
+  stop_cluster <- FALSE
+  cl <- NULL
+
+  if(!is.null(cores)) {
+    if(!requireNamespace("parallel", quietly = TRUE)) {
+      stop("parallel required if using cores input")
+    }
+    if(is.numeric(cores)) {
+      cl <- parallel::makeCluster(cores)
```

Comment:
could this approach be used? Its what I use with climateR/nwmTools to determine core usage
https://github.com/mikejohnson51/climateR/blob/c43659063f63e6ad525b29cf29321e8d924dd407/R/utility_fast_download.R#L16

Reply:
I'd rather not push the cores logic down into this function to leave it up to the script that's calling.